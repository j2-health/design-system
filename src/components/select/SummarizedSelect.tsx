import { Select, Spin, Input, SelectProps } from 'antd'
import { useState, useMemo } from 'react'
import { LoadingOutlined } from '@ant-design/icons'
import * as icons from '../icons'
import { Tag } from '../tag'
import cx from 'classnames'

import styles from './SummarizedSelect.module.css'

type Option = {
  label: string
  value: string
}

type StringSelectProps = SelectProps<string, Option>

type BaseProps = Omit<
  StringSelectProps,
  'mode' | 'value' | 'onChange' | 'options'
> & {
  searchPlaceholder?: string
  formControlPlaceholder: string
  options: Option[]
  loading?: boolean
  renderLabel: (count: number) => string
}

type SingleProps = {
  multiple?: false
  value: string
  onChange: (value: string) => void
}

type MultiProps = {
  multiple: true
  value: string[]
  onChange: (value: string[]) => void
}

export type Props = BaseProps & (SingleProps | MultiProps)

export function SummarizedSelect({
  searchPlaceholder,
  formControlPlaceholder,
  loading,
  multiple,
  value: selectedValue,
  onChange: setSelectedValue,
  options,
  renderLabel,
  ...props
}: Props) {
  const [searchValue, setSearchValue] = useState('')

  const filteredOptions = useMemo(() => {
    if (!searchValue) return options
    return options.filter((option) =>
      option.label.toLowerCase().includes(searchValue.toLowerCase())
    )
  }, [searchValue])

  const handleTagClose = (removedValue: string) => {
    if (!multiple) return
    setSelectedValue(selectedValue.filter((value) => value !== removedValue))
  }

  const popupRender = (menu: React.ReactElement) => (
    <div>
      <div className="mb-2 px-2 pb-2 border-b -mx-1 mt-1">
        <Input
          placeholder={searchPlaceholder || 'Search...'}
          value={searchValue}
          onChange={(e) => {
            setSearchValue(e.target.value)
            e.stopPropagation()
            e.preventDefault()
          }}
          onKeyDown={(e) => {
            // Prevent backspace from removing tags when typing in search input
            e.stopPropagation()
          }}
          onClick={(e) => e.stopPropagation()}
          prefix={
            <icons.MagnifyingGlassIcon size={16} className="text-gray-400" />
          }
          allowClear
        />
      </div>
      {multiple ? (
        selectedValue.length > 0 && (
          <div className="px-2 pb-2">
            <div className="flex flex-wrap gap-1">
              {selectedValue.map((value) => (
                <Tag
                  className="opacity-100 z-10"
                  key={value}
                  status="default"
                  closable
                  onClose={() => handleTagClose(value)}
                >
                  {value}
                </Tag>
              ))}
              {selectedValue}
            </div>
          </div>
        )
      ) : (
        <div className="px-2 pb-2">{selectedValue}</div>
      )}
      <div className={cx(styles.menuContainer, 'rounded-lg')}>{menu}</div>
    </div>
  )

  const handleOpenChange = (open: boolean) => {
    if (!open) {
      setSearchValue('')
    }
  }

  return (
    <Select
      showSearch={false}
      suffixIcon={
        loading ? (
          <Spin
            indicator={<LoadingOutlined spin />}
            size="small"
            data-testid="loading-spinner"
          />
        ) : (
          <icons.CaretDownIcon size={14} data-testid="caret-down" />
        )
      }
      placeholder={formControlPlaceholder}
      mode={multiple ? 'multiple' : undefined}
      value={selectedValue}
      onChange={setSelectedValue}
      maxTagCount={0}
      maxTagPlaceholder={() => renderLabel(selectedValue.length)}
      options={filteredOptions}
      popupRender={popupRender}
      onOpenChange={handleOpenChange}
      classNames={{
        root: cx('w-[200px]', styles.summarizedSelect, {
          [styles.hemisphericSelect]: props.variant != 'underlined',
        }),
        popup: {
          root: '!w-[300px]',
        },
      }}
      {...props}
    />
  )
}
