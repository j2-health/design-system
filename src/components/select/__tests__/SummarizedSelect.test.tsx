import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { SummarizedSelect } from '../SummarizedSelect'

const mockOptions = [
  { label: 'Bronze Plan', value: 'bronze' },
  { label: 'Silver Plan', value: 'silver' },
  { label: 'Gold Plan', value: 'gold' },
  { label: 'Platinum Plan', value: 'platinum' },
]

const mockGroupedOptions = [
  {
    label: 'Popular Plans',
    options: [
      { label: 'Bronze Plan', value: 'bronze' },
      { label: 'Silver Plan', value: 'silver' },
    ],
  },
  {
    label: 'Premium Plans',
    options: [
      { label: 'Gold Plan', value: 'gold' },
      { label: 'Platinum Plan', value: 'platinum' },
    ],
  },
]

describe('SummarizedSelect', () => {
  describe('Basic Rendering', () => {
    it('should render correctly with default props', () => {
      const { container } = render(
        <SummarizedSelect options={mockOptions} value="" onChange={() => {}} />
      )
      expect(container).toMatchSnapshot()
    })

    it('should apply custom class names', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          rootClassName="custom-root"
          popupClassName="custom-popup"
        />
      )
      expect(container.querySelector('.custom-root')).toBeTruthy()
    })

    it('should render with custom placeholders', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          formControlPlaceholder="Select a plan"
          searchPlaceholder="Search plans..."
        />
      )
      expect(container).toMatchSnapshot()
    })
  })

  describe('Single Selection Mode', () => {
    it('should render with selected value', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockOptions}
          value="bronze"
          onChange={() => {}}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('should call onChange with correct value when option selected', async () => {
      const mockOnChange = jest.fn()
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={mockOnChange}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const option = screen.getByText('Bronze Plan')
      await user.click(option)

      expect(mockOnChange).toHaveBeenCalledWith('bronze')
    })

    it('should display selected option correctly', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockOptions}
          value="silver"
          onChange={() => {}}
        />
      )

      expect(
        container.querySelector('.ant-select-selection-item')
      ).toHaveTextContent('Silver Plan')
    })
  })

  describe('Multiple Selection Mode', () => {
    it('should render with multiple values', () => {
      const { container } = render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['bronze', 'silver']}
          onChange={() => {}}
          renderLabel={(count) => `${count} plans selected`}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('should show custom label from renderLabel', () => {
      const { container } = render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['bronze', 'silver']}
          onChange={() => {}}
          renderLabel={(count) => `${count} plans selected`}
        />
      )

      expect(
        container.querySelector('.ant-select-selection-overflow')
      ).toHaveTextContent('2 plans selected')
    })

    it('should display tags in popup when opened', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['bronze', 'silver']}
          onChange={() => {}}
          renderLabel={(count) => `${count} selected`}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      // Tags show the option labels in the popup
      const tags = document.querySelectorAll('.ant-tag')
      expect(tags).toHaveLength(2)
      expect(tags[0]).toHaveTextContent('Bronze Plan')
      expect(tags[1]).toHaveTextContent('Silver Plan')
    })

    it('should call onChange when tag is removed', async () => {
      const mockOnChange = jest.fn()
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['bronze', 'silver']}
          onChange={mockOnChange}
          renderLabel={(count) => `${count} selected`}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      // Find and click the close button on the first tag
      const closeButton = document.querySelector('.ant-tag-close-icon')
      expect(closeButton).toBeInTheDocument()
      await user.click(closeButton as Element)

      expect(mockOnChange).toHaveBeenCalledWith(['silver'])
    })

    it('should call onChange with new array when option selected', async () => {
      const mockOnChange = jest.fn()
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['bronze']}
          onChange={mockOnChange}
          renderLabel={(count) => `${count} selected`}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const option = screen.getByText('Silver Plan')
      await user.click(option)

      expect(mockOnChange).toHaveBeenCalledWith(['bronze', 'silver'])
    })
  })

  describe('Grouped Options', () => {
    it('should render grouped options correctly', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockGroupedOptions}
          value=""
          onChange={() => {}}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('should select from grouped options', async () => {
      const mockOnChange = jest.fn()
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          options={mockGroupedOptions}
          value=""
          onChange={mockOnChange}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const option = screen.getByText('Gold Plan')
      await user.click(option)

      expect(mockOnChange).toHaveBeenCalledWith('gold')
    })
  })

  describe('Search Functionality', () => {
    it('should show search input when dropdown opens', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          searchPlaceholder="Search plans..."
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      expect(screen.getByPlaceholderText('Search plans...')).toBeInTheDocument()
    })

    it('should filter options based on search text', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect options={mockOptions} value="" onChange={() => {}} />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const searchInput = screen.getByPlaceholderText('Search...')
      await user.type(searchInput, 'Bronze')

      expect(screen.getByText('Bronze Plan')).toBeInTheDocument()
      expect(screen.queryByText('Silver Plan')).not.toBeInTheDocument()
    })

    it('should filter grouped options based on search text', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          options={mockGroupedOptions}
          value=""
          onChange={() => {}}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const searchInput = screen.getByPlaceholderText('Search...')
      await user.type(searchInput, 'Gold')

      expect(screen.getByText('Gold Plan')).toBeInTheDocument()
      expect(screen.queryByText('Bronze Plan')).not.toBeInTheDocument()
    })

    it('should clear search when dropdown closes', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect options={mockOptions} value="" onChange={() => {}} />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const searchInput = screen.getByPlaceholderText('Search...')
      await user.type(searchInput, 'Bronze')

      await user.click(document.body)
      await user.click(select)

      const newSearchInput = screen.getByPlaceholderText('Search...')
      expect(newSearchInput).toHaveValue('')
    })

    it('should prevent propagation on search input events', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect options={mockOptions} value="" onChange={() => {}} />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      const searchInput = screen.getByPlaceholderText('Search...')

      await user.type(searchInput, 'test')

      expect(searchInput).toHaveValue('test')
    })
  })

  describe('Loading State', () => {
    it('should show loading spinner when loading is true', () => {
      render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          loading
        />
      )

      expect(screen.getByTestId('loading-spinner')).toBeInTheDocument()
      expect(screen.queryByTestId('caret-down')).not.toBeInTheDocument()
    })

    it('should show caret icon when not loading', () => {
      render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          loading={false}
        />
      )

      expect(screen.getByTestId('caret-down')).toBeInTheDocument()
      expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument()
    })

    it('should render with loading state', () => {
      const { container } = render(
        <SummarizedSelect
          options={mockOptions}
          value=""
          onChange={() => {}}
          loading
        />
      )
      expect(container).toMatchSnapshot()
    })
  })

  describe('Edge Cases', () => {
    it('should handle empty options array', () => {
      const { container } = render(
        <SummarizedSelect options={[]} value="" onChange={() => {}} />
      )
      expect(container).toMatchSnapshot()
    })

    it('should handle options with undefined labels', () => {
      const optionsWithUndefinedLabel = [
        { label: undefined, value: 'test' },
        { label: 'Valid Label', value: 'valid' },
      ]

      const { container } = render(
        <SummarizedSelect
          options={optionsWithUndefinedLabel}
          value=""
          onChange={() => {}}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('should throw error when value is not string in single mode', () => {
      const mockOnChange = jest.fn()

      const TestComponent = () => {
        const handleChange = (val: string) => {
          mockOnChange(val)
        }

        return (
          <SummarizedSelect
            options={mockOptions}
            value=""
            onChange={handleChange}
          />
        )
      }

      expect(() => render(<TestComponent />)).not.toThrow()
    })

    it('should throw error when value is not array in multiple mode', () => {
      const mockOnChange = jest.fn()

      const TestComponent = () => {
        const handleChange = (val: string[]) => {
          mockOnChange(val)
        }

        return (
          <SummarizedSelect
            multiple
            options={mockOptions}
            value={[]}
            onChange={handleChange}
            renderLabel={(count) => `${count} selected`}
          />
        )
      }

      expect(() => render(<TestComponent />)).not.toThrow()
    })

    it('should handle missing option in valueToLabel function', async () => {
      const user = userEvent.setup()

      render(
        <SummarizedSelect
          multiple
          options={mockOptions}
          value={['nonexistent']}
          onChange={() => {}}
          renderLabel={(count) => `${count} selected`}
        />
      )

      const select = screen.getByRole('combobox')
      await user.click(select)

      expect(screen.getByText('nonexistent')).toBeInTheDocument()
    })
  })
})
