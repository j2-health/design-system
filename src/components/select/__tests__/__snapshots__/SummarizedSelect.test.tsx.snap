// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummarizedSelect Basic Rendering should render correctly with default props 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Basic Rendering should render with custom placeholders 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Edge Cases should handle empty options array 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Edge Cases should handle options with undefined labels 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Grouped Options should render grouped options correctly 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Loading State should render with loading state 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title=""
        />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <div
        aria-busy="true"
        aria-live="polite"
        class="ant-spin ant-spin-sm ant-spin-spinning css-dev-only-do-not-override-os2fvf"
        data-testid="loading-spinner"
      >
        <span
          aria-label="loading"
          class="anticon anticon-loading anticon-spin ant-spin-dot"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </span>
      </div>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Multiple Selection Mode should render with multiple values 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-multiple ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <div
          class="ant-select-selection-overflow"
        >
          <div
            class="ant-select-selection-overflow-item ant-select-selection-overflow-item-rest"
            style="opacity: 1;"
          >
            <span
              class="ant-select-selection-item"
              title="2 plans selected"
            >
              <span
                class="ant-select-selection-item-content"
              >
                2 plans selected
              </span>
            </span>
          </div>
          <div
            class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
            style="opacity: 1;"
          >
            <div
              class="ant-select-selection-search"
              style="width: 0px;"
            >
              <input
                aria-autocomplete="list"
                aria-controls="rc_select_TEST_OR_SSR_list"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="rc_select_TEST_OR_SSR_list"
                autocomplete="off"
                class="ant-select-selection-search-input"
                id="rc_select_TEST_OR_SSR"
                readonly=""
                role="combobox"
                style="opacity: 0;"
                type="search"
                unselectable="on"
                value=""
              />
              <span
                aria-hidden="true"
                class="ant-select-selection-search-mirror"
              >
                 
              </span>
            </div>
          </div>
        </div>
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SummarizedSelect Single Selection Mode should render with selected value 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined summarizedSelect hemisphericSelect css-dev-only-do-not-override-os2fvf ant-select-single ant-select-show-arrow"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-wrap"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="Bronze Plan"
        >
          Bronze Plan
        </span>
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        data-testid="caret-down"
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</div>
`;
